### ISSUE TITLE
[REVIEW]: pulse

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1539]
**Submitting author:** @finsberg (<a href="http://orcid.org/0000-0003-3766-2393">Henrik Nicolay Topnes Finsberg</a>)
**Repository:** <a href="https://github.com/finsberg/pulse" target ="_blank">https://github.com/finsberg/pulse</a>
**Version:** 2019.0
**Editor:** @trallard
**Reviewer:** @melund, @trallard 
**Archive:** <a href="https://doi.org/10.5281/zenodo.3376811" target="_blank">10.5281/zenodo.3376811</a>

## Status

[![status](http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40/status.svg)](http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40"><img src="http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40/status.svg)](http://joss.theoj.org/papers/9abee735e6abadabe9252d5fcc84fd40)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/finsberg/pulse" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@melund & @trallard , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @trallard know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/finsberg/pulse">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 2019.0
- [x] **Authorship:** Has the submitting author (@finsberg) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/finsberg/pulse">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 2019.0
- [x] **Authorship:** Has the submitting author (@finsberg) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @melund, @trallard  it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by trallard** [comment]
Hi @melund  your reviewer checklist is at the top 👆🏼 thanks a lot for your time and effort. 
Let's use this issue to keep the review alive and use the submitting repo for any longer reviews/discussions. 

As per usual if you run into any problems feel free to ping me here!

**Comment by melund** [comment]
Thanks, @trallard. I may be a little while before I can get to this. I am going on sommer holiday on Friday. And I will most likely be away from my computer for at least two weeks. 

I am sorry, but I am just too busy before I leave to get this done. I will look at it when I get back :)

**Comment by trallard** [comment]
Hi @melund just checking how you are doing? I am planning to move this forward this week - early the next

**Comment by melund** [comment]
@trallard I am traveling in Asia and I only have my phone. I am back in a week and then I will try to get look at this.

**Comment by trallard** [comment]
@whedon remind @melund in 2 weeks

**Comment by whedon** [comment]
Reminder set for @melund in 2 weeks

**Comment by finsberg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by finsberg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by whedon** [comment]
:wave: @melund, please update us on how your review is going.

**Comment by melund** [comment]
I am done with my review. I think pulse more than qualifies for JOSS. It is a really good example of how to structure an open-source software project. With the recent addition of conda-forge package, I am also sure that the library will continue to work in the future. 

All in all, nice work @finsberg  👍

**Comment by trallard** [comment]
Thanks @melund 
I just noticed that the item for the license has not been checked on your list but I just double checked and this is in compliance. So @finsberg  I am going to move to the following editorial activites. 

Can you please make sure to perform the following actions:
1. Generate a new DOI for the software and past it here
2. Make sure that the package version is up to date and report here: if needed make a new release and report it here
3. Do a last review of the paper

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by trallard** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.1007/s10439-012-0593-5 may be missing for title: A novel rule-based algorithm for assigning myocardial fiber orientation to computational heart models
- https://doi.org/10.1016/j.jocs.2017.07.013 may be missing for title: Estimating cardiac contraction through high resolution data assimilation of a personalized mechanical model
- https://doi.org/10.1002/cnm.2982 may be missing for title: Efficient estimation of personalized biventricular mechanical function employing gradient-based optimization
- https://doi.org/10.1016/j.cam.2012.10.034 may be missing for title: A computational method to assess the in vivo stresses and unloaded configuration of patient-specific blood vessels

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by finsberg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by finsberg** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1007/s10439-012-0593-5 is OK
- 10.1016/j.jocs.2017.07.013  is OK
- 10.1002/cnm.2982 is OK
- 10.1016/j.cam.2012.10.034 is OK
- 10.1098/rspa.2015.0641 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by finsberg** [comment]
1. @trallard How do you want me do generate a DOI for the software? [Zenodo](https://zenodo.org)?
2. Package version number [2019.0](https://github.com/finsberg/pulse/releases/tag/v2019.0) is up to date.
3. I have done final review of the paper, and made a few corrections. The last article proof looks good. I also added the DOIs in the bib-file.

**Comment by trallard** [comment]
Hey @finsberg  you can archive the software in Zenodo and add here the DOI for it

**Comment by finsberg** [comment]
DOI: 10.5281/zenodo.3376811

**Comment by melund** [comment]
@trallard I forgot to check the license box. Everything is fine now. Good work @finsberg

**Comment by arfon** [comment]
:wave: @trallard - I think this paper is waiting on you to make your last checks here.

**Comment by trallard** [comment]
Hi everyone thanks for the ping! I will finalise this today's evening

**Comment by trallard** [comment]
@whedon check DOI

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by trallard** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1007/s10439-012-0593-5 is OK
- 10.1016/j.jocs.2017.07.013  is OK
- 10.1002/cnm.2982 is OK
- 10.1016/j.cam.2012.10.034 is OK
- 10.1098/rspa.2015.0641 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by trallard** [comment]
@whedon set 10.5281/zenodo.3376811 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3376811" target="_blank">10.5281/zenodo.3376811</a> is the archive.

**Comment by trallard** [comment]
@whedon set 2019.0 as version

**Comment by whedon** [comment]
OK. 2019.0 is the version.

**Comment by trallard** [comment]
@finsberg  I have completed my editorial checks and I am willing to recommend this for publication 🎉

@melund thanks for your contribution as a reviewer 🙌🏼

@openjournals/jose-eics  I am passing this submission to you for acceptance

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by danielskatz** [comment]
👋 @trallard - fwiw, the accepted label will be automatically attached during the final processing

**Comment by trallard** [comment]
Thanks for the heads up. I have been following the instructions at https://joss.readthedocs.io/en/latest/editing.html#after-acceptance where it says the editor should add the `accepted` label

**Comment by danielskatz** [comment]
👋 @finsberg - please merge in the changes in https://github.com/finsberg/pulse/pull/8 and https://github.com/finsberg/pulse/pull/9 or let me know what you disagree with

**Comment by finsberg** [comment]
@danielskatz Done

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01539/joss.01539/10.21105.joss.01539.pdf)

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/979

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/979, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1176537599143481352) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/980
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01539](https://doi.org/10.21105/joss.01539)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @melund for reviewing and @trallard for both reviewing and editing.
And congratulations to @finsberg

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01539/status.svg)](https://doi.org/10.21105/joss.01539)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01539">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01539/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01539/status.svg
   :target: https://doi.org/10.21105/joss.01539
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01539/status.svg)](https://doi.org/10.21105/joss.01539)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss
